### PR TITLE
Models testing with rspec

### PR DESCRIPTION
**Pull request by nhs3108**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -57,4 +57,5 @@ group :development, :test do
   gem "rspec-rails", "~> 3.0"
   gem "factory_girl_rails", "~> 4.0"
   gem "rspec-activemodel-mocks"
+  gem 'shoulda-matchers'
```
Comment (ttuan) [inline Gemfile]: use ""

File: spec/models/course_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      before{course.destroy}
+      it{expect(course.subjects.count).to eq 0}
+    end
+  end
+
+  describe "association" do
+    it{should have_many(:course_subjects)}
+    it{should have_many(:user_courses)}
+    it{should have_many(:supervisor_courses)}
+    it{should have_many(:subjects).through(:course_subjects)}
```
Comment (hienbx94) [inline spec/models/course_spec.rb]: remove ()

File: spec/models/course_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    }
+    it{expect(course.subjects.count).to eq 2}
+    context "when deleting the course" do
+      before{course.destroy}
+      it{expect(course.subjects.count).to eq 0}
+    end
+  end
+
+  describe "association" do
+    it{should have_many(:course_subjects)}
```
Comment (hienbx94) [inline spec/models/course_spec.rb]: remove ()

File: spec/models/course_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    it{should have_many(:user_courses)}
```
Comment (hienbx94) [inline spec/models/course_spec.rb]: Nhu tren

File: spec/models/course_spec.rb (hunk 2 above)
Comment (thiennv57) [inline spec/models/course_spec.rb]: remove ()

File: spec/models/course_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
+require "spec_helper"
```
Comment (ttuan) [inline spec/models/course_spec.rb]: can remove

File: spec/models/course_spec.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      it{expect(course.subjects.count).to eq 0}
+    end
+  end
+
+  describe "association" do
+    it{should have_many(:course_subjects)}
+    it{should have_many(:user_courses)}
+    it{should have_many(:supervisor_courses)}
+    it{should have_many(:subjects).through(:course_subjects)}
+    it{should have_many(:users).through(:user_courses)}
```
Comment (thiennv57) [inline spec/models/course_spec.rb]: remove ()

**Pull request merged**